src/hotspot/share/code/codeBlob.hpp

```diff
@@ -108,8 +108,6 @@ class CodeBlob {
   int      _relocation_size;       // size of relocation (could be bigger than 64Kb)
```

Comment:
For offsets into the external mutable/immutable data, we could reduce codecache footprint further by moving these into a a header section of the external data block.  That also allows those blocks to be self-describing, which could help with error reporting or debugging.

Reply:
Sounds reasonable. But the downside is that in this case the oops iterator needs an additional load (nmethod->mutable_data->oop_size) to check if the oops list is empty.